### ISSUE TITLE
Deployment failed

### ISSUE THREAD

**Issue by yaowilli** [Azure-Samples/chat-with-your-data-solution-accelerator#1195]
While deploying this accelerator (https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/main/docs/LOCAL_DEPLOYMENT.md)  performing a local deployment i am getting the following error which does not provide much details 
{
  'code': 'DeploymentFailed',
  'target': '/subscriptions/f413fa15-c612-4fd4-85e8-949dd818aeee/resourceGroups/rg-rg-ipas-002/providers/Microsoft.Resources/deployments/backend-qrwpczu6q6ozk',
  'message': 'At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.',
  'details': [
    {
      'code': 'BadRequest',
      'target': '/subscriptions/f413fa15-c612-4fd4-85e8-949dd818aeee/resourceGroups/rg-rg-ipas-002/providers/Microsoft.Web/sites/backend-qrwpczu6q6ozk/host/default/functionKeys/clientKey',
      'message': '{\\r\\n  \\'Code\\': \\'BadRequest\\',\\r\\n  \\'Message\\': \\'Encountered an error (InternalServerError) from host runtime.\\',\\r\\n  \\'Target\\': null,\\r\\n  \\'Details\\': [\\r\\n    {\\r\\n      \\'Message\\': \\'Encountered an error (InternalServerError) from host runtime.\\'\\r\\n    },\\r\\n    {\\r\\n      \\'Code\\': \\'BadRequest\\'\\r\\n    },\\r\\n    {\\r\\n   
![image](https://github.com/user-attachments/assets/7ff9199d-b928-4de2-8432-2e84d9b6d7db)
   \\'ErrorEntity\\': {\\r\\n        \\'Code\\': \\'BadRequest\\',\\r\\n        \\'Message\\': \\'Encountered an error (InternalServerError) from host runtime.\\'\\r\\n      }\\r\\n    }\\r\\n  ],\\r\\n  \\'Innererror\\': null\\r\\n}'
    }
  ]

instructions followed: https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/main/docs/LOCAL_DEPLOYMENT.md

**Comment by Roopan-Microsoft** [comment]
Hi @yaowilli - Thanks for reaching out to us.

Can you please let us know more details on below points, while we are investigating this issue?
- Did you used Dev Container for Local Deployment?
- Are you getting this issue post you executed **_azd up_**?
- Did you try redeploying?
- Also, have you tried one click deployment? If yes are you facing similar issue there as well?

**Comment by yaowilli** [comment]
Hello


  *   Did you used Dev Container for Local Deployment? [William]: using docker container
  *   Are you getting this issue post you executed azd up?  [William]: Yes
  *   Did you try redeploying? [William]: did try several times
  *   Also, have you tried one click deployment? If yes are you facing similar issue there as well? [William]: tried one click deployment but the template is incorrect I do not to the deployment

**Comment by yaowilli** [comment]
![activity_log_error](https://github.com/user-attachments/assets/ccad8f93-9923-4de3-be7a-58fdc8807ad3)

**Comment by yaowilli** [comment]
the error occurs during the execution of azd deploy inside azd up wrapper

**Comment by yaowilli** [comment]
Would you please have any suggestion ?
This help page here is not helpful https://learn.microsoft.com/en-us/azure/azure-resource-manager/troubleshooting/common-deployment-errors

**Comment by AntonFLB** [comment]
Any news on this topic, i am facing same issue when deploy the pack into our azure environment.

**Comment by yaowilli** [comment]
![image](https://github.com/user-attachments/assets/eee28663-3c47-407f-8244-71b57803d2e8)

**Comment by yaowilli** [comment]
I was able to fix the issue

**Comment by Vinay-Microsoft** [comment]
Hi @yaowilli What did you do to fix this Issue?

**Comment by yaowilli** [comment]
add the storage blob data contributor and storage data reader roles to the managed identity of the function app and ensure that when deploying solution you have in bicep file
 @allowed(['Enabled', 'Disabled'])
param publicNetworkAccess string = 'Enabled'

**Comment by ross-p-smith** [comment]
Hi @yaowilli - do you fancy putting together a Pull Request and I will review

**Comment by yaowilli** [comment]
sure how ?

**Comment by atockar** [comment]
Hi, I am still getting this issue. I have tried to deploy using the one click deployment and it fails in the same place, 
![Screenshot 2024-08-30 at 4 35 42 pm](https://github.com/user-attachments/assets/8529fc9a-3a32-4084-b5cb-639bd8a5ee2f)

I can see the bicep file has been updated in the code, and I added the storage blob data contributor role for the storage account to the function app:
> add the storage blob data contributor and storage data reader roles to the managed identity of the function app and ensure that when deploying solution you have in bicep file @Allowed(['Enabled', 'Disabled']) param publicNetworkAccess string = 'Enabled'

However when I redeploy this section it asks for the clientKey, which I thought was generated. Any ideas?
